File: poultry-pal-service/src/main/java/co/za/hlaluko/dynamics/poultry/pal/model/dto/Mortality.java
package co.za.hlaluko.dynamics.poultry.pal.model.dto;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Mortality {
  private String id;
  private Date dateOccurred;
  private int numberOfDeaths;
  private String reason;
  private String recordedBy;
  private Date createdDate;
}

File: poultry_pal_plus_app/lib/models/mortality.dart
class Mortality {
  final String id;
  final String dateOccurred;
  final int numberOfDeaths;
  final String reason;
  final String recordedBy;
  final String createdDate;

  Mortality({
    required this.id,
    required this.dateOccurred,
    required this.numberOfDeaths,
    required this.reason,
    required this.recordedBy,
    required this.createdDate,
  });

  factory Mortality.fromJson(Map<String, dynamic> json) {
    return Mortality(
      id: json['id'],
      dateOccurred: json['dateOccurred'],
      numberOfDeaths: json['numberOfDeaths'],
      reason: json['reason'],
      recordedBy: json['recordedBy'],
      createdDate: json['createdDate'],
    );
  }
}
